indra/db/client.py

```diff
@@ -415,21 +417,21 @@ def get_statements(clauses, count=1000, do_stmt_count=False, db=None,
         # Populate the supports/supported by fields.
         if with_support:
             logger.info("Populating support links.")
-            support_links = db.filter_query(
+            support_links = db.select_all(
                 [db.PASupportLinks.supported_mk_hash,
                  db.PASupportLinks.supporting_mk_hash],
                 or_(db.PASupportLinks.supported_mk_hash.in_(stmt_dict.keys()),
                     db.PASupportLinks.supporting_mk_hash.in_(stmt_dict.keys()))
-                ).distinct().yield_per(count)
-            for supped_hash, supping_hash in support_links:
+                )
+            for supped_hash, supping_hash in set(support_links):
+                if supped_hash == supping_hash:
+                    assert False, 'Self-support found on-load.'
                 supped_stmt = stmt_dict.get(supped_hash,
                                             Unresolved(shallow_hash=supped_hash))
                 supping_stmt = stmt_dict.get(supping_hash,
                                              Unresolved(shallow_hash=supping_hash))
-                if not isinstance(supped_stmt, str):
-                    supped_stmt.supported_by.append(supping_stmt)
-                if not isinstance(supping_stmt, str):
-                    supping_stmt.supports.append(supped_stmt)
+                supped_stmt.supported_by.append(supping_stmt)
+                supping_stmt.supports.append(supped_stmt)
```

Comment:
All of the changes to this file are to either simplify the code or improve logging, and also to add a check for self-support.